### PR TITLE
Parse JSON reports of google test

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Use `--gtest_output='json:'` to parse unit test results.

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 388c47a59a3f858a06922f4ec2c5dcf976931eee with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54922/388c47a59a3f858a06922f4ec2c5dcf976931eee/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: tests/ci/unit_tests_check.py [hunk 1]
```diff
@@ -43,67 +45,120 @@ def get_test_name(line):
...
-
-    if failed_counter != 0:
-        status = "failure"
+    SEGFAULT = "Segmentation fault. "
+    SIGNAL = "Exit on signal. "
+    for suite in report["testsuites"]:
+        suite_name = suite["name"]
+        for test_case in suite["testsuite"]:
+            case_name = test_case["name"]
+            test_time = float(test_case["time"][:-1])
```
Comment (antaljanosbenjamin) [inline tests/ci/unit_tests_check.py:123]: Why the `[:-1]` part? Isn't `google.protobuf.Duration` a single struct? As I see it is not a repeated field here and the https://github.com/protocolbuffers/protobuf/blob/a30d71d873f8f91dca230c204a6757c7a4cbaed7/src/google/protobuf/duration.proto#L102-L115C2 (seconds and nanos). If I am missing something here, then sorry.

File: tests/ci/unit_tests_check.py [hunk 2]
```diff
@@ -43,67 +45,120 @@ def get_test_name(line):
...
+            raw_logs = None
+            if "failures" in test_case:
+                raw_logs = ""
+                for failure in test_case["failures"]:
+                    raw_logs += failure["failure"]
```
Comment (antaljanosbenjamin) [inline tests/ci/unit_tests_check.py:128]: Based on the protobuf above the `Failure` struct has a `failures` field, but not `failure`.
```suggestion
                    raw_logs += failure["failures"]
```

**Comment by Felixoid** [comment]
@antaljanosbenjamin here's the report output I've used to develop the thing 
[unit-tests-report.txt](https://github.com/ClickHouse/ClickHouse/files/12700925/unit-tests-report.txt)

**Comment by antaljanosbenjamin** [comment]
I see the JSON has important differences with the proto mentioned in the file. Maybe alter the proto to be an accurate representation of the JSON? Otherwise it is confusing (as it confused me during review).

**Comment by Felixoid** [comment]
Maybe, update the framework and see if the scheme is changed?

https://github.com/google/googletest/compare/71140c3..main

**Comment by Felixoid** [comment]
I think, the message itself is very outdated, and it'd be useful to update it in the documentation too https://github.com/google/googletest/discussions/4378

**Comment by Felixoid** [comment]
But it works! How cool is that? https://s3.amazonaws.com/clickhouse-test-reports/54922/aef370ca7c08946476d1eba83bcd0d42a16f783b/unit_tests__release_.html

**Comment by Felixoid** [comment]
I've added a note, that the scheme is wrong but copied from the documentation.

I think the best would be merging it, and solving inconsistency in the documentation separately

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by Felixoid** [comment]
image

Let's gooo!

**Comment by Felixoid** [comment]
Oh, no.. there's one thing that remains. No `All tests` in the end of the report

**Comment by Felixoid** [comment]
image
we're good to go

**Pull request merged**